bundles/ranvier-quests/lib/BountyGoal.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const QuestGoal = require('../../../src/QuestGoal');
+
+/**
+ * A quest goal requiring the player to locate and/or return an NPC.
+ */
+class BountyGoal extends QuestGoal {
+  constructor(quest, config, player) {
+    config = Object.assign({
+      title: 'Locate NPC',
+      npc: null, // NPC ID to capture
+      home: null // Area ID to return to
+    }, config);
+
+    super(quest, config, player);
+
+    this.state = {
+      found: false,
+      delivered: false
+    };
+
+    this.on('enterRoom', this._enterRoom);
+  }
+
+  getProgress() {
+    // Has target been located?
+    let percent = this.state.found ? 50 : 0;
+    
+    if (this.config.home) {
+      // Has target been returned home?
+      percent += this.state.delivered ? 50 : 0;
+    } else {
+      // No return location necessary.
+      percent += 50;
+    }
+    
+    const display = this.state.found ? 'Complete' : 'Not Complete';
+    return { percent, display };
+  }
+  
+  _enterRoom(room) {
+    if (this.state.found) {
+      if (room.entityReference == this.config.home) {
+        // Check if we have taken the NPC home
+        this.state.delivered = true;
+      }
+      this.emit('progress', this.getProgress());      
+    } else {
+      let located = false;
+      const goalnpcid = this.config.npc;
+      if (goalnpcid != null) {
```

Comment:
Should there be an error if goalNpcId is null? Also, I would suggest using `goalNpcId !== null` for strict equality... slightly pedantic but could prevent weird bugs that might otherwise be hard to track down.

Reply:
yeah I think what I could do is making `QuestGoal` have a static `validate()` method that is called on bootup so the server won't start if you have invalid quest configurations. Like this goal could validate that it has a goal NPC

Reply:
@seanohue You are correct, there's a problem if the `goalnpcid` is null. I'll update that with a `!== null` check until @shawncplus can work in a `validate()` method.